src/core/server/ui_settings/settings/date_formats.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import moment from 'moment-timezone';
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { UiSettingsParams } from '../../../types';
+
+export const getDateFormatSettings = (): Record<string, UiSettingsParams> => {
+  const weekdays = moment.weekdays().slice();
+  const [defaultWeekday] = weekdays;
+
+  const timezones = [
+    'Browser',
+    ...moment.tz
+      .names()
+      // We need to filter out some time zones, that moment.js knows about, but Elasticsearch
+      // does not understand and would fail thus with a 400 bad request when using them.
+      .filter((tz) => !['America/Nuuk', 'EST', 'HST', 'ROC', 'MST'].includes(tz)),
+  ];
+
+  return {
+    dateFormat: {
+      name: i18n.translate('kbn.advancedSettings.dateFormatTitle', {
+        defaultMessage: 'Date format',
+      }),
+      value: 'MMM D, YYYY @ HH:mm:ss.SSS',
+      description: i18n.translate('kbn.advancedSettings.dateFormatText', {
+        defaultMessage: 'When displaying a pretty formatted date, use this {formatLink}',
+        description:
+          'Part of composite text: kbn.advancedSettings.dateFormatText + ' +
+          'kbn.advancedSettings.dateFormat.optionsLinkText',
+        values: {
+          formatLink:
+            '<a href="https://momentjs.com/docs/#/displaying/format/" target="_blank" rel="noopener noreferrer">' +
+            i18n.translate('kbn.advancedSettings.dateFormat.optionsLinkText', {
+              defaultMessage: 'format',
+            }) +
+            '</a>',
+        },
+      }),
+      schema: schema.string(),
```

Comment:
AFAIK there is no way to validate a moment date format. We can't do better than just `string` here unfortunately...